ambari-common/src/main/python/resource_management/libraries/functions/stack_features.py

```diff
@@ -102,24 +103,28 @@ def get_stack_feature_version(config):
                   and commandParams from.
   :return: the version to use when checking stack features.
   """
-  from resource_management.libraries.functions.default import default
+  from resource_management.libraries.script.script import Script
+  from resource_management.libraries.execution_command.execution_command import ExecutionCommand
+
+  execution_command = Script.get_execution_command()
 
-  if "stackSettings" not in config or "commandParams" not in config:
+  if "stackSettings" not in config and "commandParams" not in config:
     raise Fail("Unable to determine the correct version since stackSettings and commandParams were not present in the configuration dictionary")
 
   # should always be there
   # Actually not always, i.e if we restart zookeeper service and no stack_version is included in command.json
-  stack_version = default("/stackSettings/stack_version", None)
+  stack_version = execution_command.get_mpack_version()
 
   # something like 2.4.0.0-1234; represents the version for the command
   # (or None if this is a cluster install and it hasn't been calculated yet)
   # this is always guaranteed to be the correct version for the command, even in
   # upgrade and downgrade scenarios
-  command_version = default("/commandParams/version", None)
-  command_stack = default("/commandParams/target_stack", None)
+  command_version = execution_command.get_new_mpack_version_for_upgrade()
+  # TODO we may need add this to execution_command lib
+  command_stack = execution_command.get_value("commandParams/target_stack", None)
```

Comment:
This is probably all going away anyway ...

Reply:
Yes, I think it may be obsolete, but at this time I just make it work.